construct/core.py

```diff
@@ -524,10 +524,12 @@ def _parse(self, stream, context):
         except ConstructError:
             if c < self.mincount:
                 raise RangeError("expected %d to %d, found %d" %
-                    (self.mincount, self.maxcout, c), sys.exc_info()[1])
+                    (self.mincount, self.maxcout, c))
             stream.seek(pos)
         return obj
     def _build(self, obj, stream, context):
+        if (not hasattr(obj, '__len__')) or hasattr(obj, 'get'):
```

Comment:
Test obj not dict for is instance?

-- Arkadiusz Bulski --

26 sie 2016 2:48 AM "Benjamin Riggs" <notifications@github.com> napisał(a):

> In construct/core.py
> <https://github.com/construct/construct/pull/79#discussion_r76350080>:
>
> >              stream.seek(pos)
> >          return obj
> >      def _build(self, obj, stream, context):
> > +        if (not hasattr(obj, '__len__')) or hasattr(obj, 'get'):
>
> isinstance(collections.abc.MutableMapping, dict) == false
> isinstance(dict, collections.abc.MutableMapping) == false
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/construct/construct/pull/79/files/7e0a04c9be917e814b6ff2fc3fbe4bb591d52f0d#r76350080>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFIubsKHQuRSRdqStUc96ADB4tUKhyx3ks5qjjfygaJpZM4JsUOg>
> .
>